File: flask_app/app/models/notifications.py
from app.models.base import BaseModel, db
from app.constants import NotificationType

class Notifications(BaseModel):
    __tablename__ = 'notifications'

    user_id  = db.Column(db.String(36), nullable=False)
    is_read = db.Column(db.Boolean, default=False)
    notification_type  = db.Column(db.Enum(NotificationType), nullable=False)
    title  = db.Column(db.String(50), nullable=False)
    message  = db.Column(db.Text, nullable=False)
    details = db.Column(db.JSON, nullable=True)

    def __repr__(self):
        return f"<Notifications {self.title}>"

File: java_membership_api/src/main/java/com/example/membership_api/model/Notifications.java
package com.example.membership_api.model;

import com.example.membership_api.constants.NotificationType;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Table;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Data
@Entity
@Table(name = "notifications")
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
public class Notifications extends BaseModel {

    @Column(name = "user_id", nullable = false, length = 36)
    private String userId;

    @Column(name = "is_read", nullable = false)
    private boolean isRead = false;

    @Enumerated(EnumType.STRING)
    @Column(name = "notification_type", nullable = false, length = 50)
    private NotificationType notificationType;

    @Column(name = "title", nullable = false, length = 50)
    private String title;

    @Column(name = "message", nullable = false, columnDefinition = "TEXT")
    private String message;

    @Column(name = "details", columnDefinition = "JSON")
    private String details;  // JSON stored as String

}
